decoder/src/log/mod.rs

```diff
@@ -119,16 +119,19 @@ impl<'a> DefmtRecord<'a> {
 /// The caller has to provide a `should_log` closure that determines whether a log record should be
 /// printed.
 ///
-/// If `always_include_location` is `true`, a second line containing location information will be
-/// printed for *all* records, not just for defmt frames (defmt frames always get location info
-/// included if it is available, regardless of this setting).
+/// If `include_location` is `true`, a second line containing location information will be
+/// printed for defmt frames.
+///
+/// If `always_include_location` is also `true`, the second line containing location information
+/// will be printed for *all* records and not just for defmt frames.
 pub fn init_logger(
+    include_location: bool,
```

Comment:
Looking at it I think `json` should be `mode: Mode` instead.

```rust
enum Mode {
    Json,
    Pretty,
}
```

But that is not really your concern 😆

Reply:
That makes a lot of sense. Before the bool was okay, since it was just two variants. But now I feel that the enum is the better way to go.

The `Mode` is also totally not relevant to this PR, I just thought about it while reviewing.

Can you please change to use the `enum IncludeLocation`?

Reply:
Sure